geomstats/geometry/poincare_ball.py

```diff
@@ -60,6 +66,132 @@ def belongs(self, point, tolerance=TOLERANCE):
         """
         return gs.sum(point**2, axis=-1) < (1 - tolerance)
 
+    @staticmethod
+    def pdf(data, means, variances, norm_func, metric):
+        """Return the probability density function.
+
+        Parameters
+        ----------
+        data : array-like, shape=[n_samples, dim]
+            Points for which the GMM probability density is computed.
+        means : array-like, shape=[n_gaussian, dim]
+            Means of each component of the GMM.
+        variances : array-like, shape=[n_gaussian]
+            Variances of each component of the GMM.
+        norm_func: : function
+            Normalisation factor function.
+        metric : function
+            Distance function associated with the used metric.
+
+        Returns
+        -------
+        result : array-like, shape=[n_samples, n_gaussian]
+            Probability density function computed at each data
+            sample and for each component of the GMM.
+        """
+        data_length, _, n_gaussian = data.shape + (means.shape[0],)
+
+        data_expanded = gs.expand_dims(data, 1)
+        data_expanded = gs.repeat(data_expanded, n_gaussian, axis=1)
+
+        means_expanded = gs.expand_dims(means, 0)
+        means_expanded = gs.repeat(means_expanded, data_length, axis=0)
+
+        variances_expanded = gs.expand_dims(variances, 0)
+        variances_expanded = gs.repeat(variances_expanded, data_length, 0)
+
+        variances_flatten = variances_expanded.flatten()
+        data_flatten = gs.reshape(data_expanded,
```

Comment:
Let's adapt to our convention for indent:
https://geomstats.github.io/contributing.html#coding-style-guidelines
See: "10. If you need several lines for a function call, use the syntax..."